### PR TITLE
Default setting of JFrog CLI relevant environment variables

### PR DESCRIPTION
**Pull request by gailazar300**
- [x] This pull request is created in the [jfrog/jenkins-jfrog-plugin](https://github.com/jfrog/jenkins-jfrog-plugin) repository.

Set 4 environment variables with a default value (in case they are empty):
1. JFROG_CLI_DEPENDENCIES_DIR: 
    By doing this, we avoid downloading dependencies separately for each job in its temporary Jfrog home directory.
2. JFROG_CLI_BUILD_NAME & JFROG_CLI_BUILD_NUMBER
3. JFROG_CLI_BUILD_URL

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/jfrog/JfStep.java [hunk 1]
```diff
@@ -100,6 +96,46 @@ public void perform(@NonNull Run<?, ?> run, @NonNull FilePath workspace, @NonNul
         }
     }
 
+    /**
+     * Configure all JFrog relevant environment variables and all servers (if they haven't been configured yet).
+     *
+     * @param run             running as a part of a specific build
```
Comment (eyalbe4) [inline src/main/java/io/jenkins/plugins/jfrog/JfStep.java:102]: ...as a part... --> ...as part...

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: src/main/java/io/jenkins/plugins/jfrog/JfStep.java [hunk 2]
```diff
@@ -100,6 +96,46 @@ public void perform(@NonNull Run<?, ?> run, @NonNull FilePath workspace, @NonNul
...
+            env.put(JFROG_CLI_BUILD_NAME, env.get("JOB_NAME"));
+        }
+        if (env.get(JFROG_CLI_BUILD_NUMBER) == null) {
+            // Setting Jenkins build number as the default build-info number.
+            env.put(JFROG_CLI_BUILD_NUMBER, env.get("BUILD_NUMBER"));
+        }
+        if (env.get(JFROG_CLI_BUILD_URL) == null) {
+            // Setting the specific build URL.
+            env.put(JFROG_CLI_BUILD_URL, env.get("BUILD_URL"));
+        }
```
Comment (yahavi) [inline src/main/java/io/jenkins/plugins/jfrog/JfStep.java:126]: We can remove code duplication by doing the following:
```suggestion
        // Setting Jenkins job name as the default build-info name.
        env.putIfAbsent(JFROG_CLI_BUILD_NAME, env.get("JOB_NAME"));
        // Setting Jenkins build number as the default build-info number.
        env.putIfAbsent(JFROG_CLI_BUILD_NUMBER, env.get("BUILD_NUMBER"));
        // Setting the specific build URL.
        env.putIfAbsent(JFROG_CLI_BUILD_URL, env.get("BUILD_URL"));
```

File: src/main/java/io/jenkins/plugins/jfrog/JfStep.java [hunk 3]
```diff
@@ -100,6 +96,46 @@ public void perform(@NonNull Run<?, ?> run, @NonNull FilePath workspace, @NonNul
...
+        // Set up a temporary Jfrog CLI home directory for a specific run.
+        FilePath jfrogHomeTempDir = Utils.createAndGetJfrogCliHomeTempDir(workspace, String.valueOf(run.getNumber()));
+        env.put(JFROG_CLI_HOME_DIR, jfrogHomeTempDir.getRemote());
```
Comment (yahavi) [inline src/main/java/io/jenkins/plugins/jfrog/JfStep.java:129]: Shouldn't we check for the absence of `JFROG_CLI_HOME_DIR` before setting it?

File: src/main/java/io/jenkins/plugins/jfrog/JfrogInstallation.java [hunk 4]
```diff
@@ -49,6 +54,17 @@ public void buildEnvVars(EnvVars env) {
...
+        if (env.get(JFROG_CLI_DEPENDENCIES_DIR) == null) {
+            // Jfrog CLI dependencies directory is a sibling of all the other tools directories.
+            // By doing this, we avoid downloading dependencies separately for each job in its temporary Jfrog home directory.
+            Path path = Paths.get(home).getParent();
+            if (path != null) {
+                env.put(JFROG_CLI_DEPENDENCIES_DIR, path.resolve(JfrogDependenciesDirName).toString());
+            }
+        }
+        if (env.get(JFROG_CLI_USER_AGENT) == null) {
+            env.put(JFROG_CLI_USER_AGENT, "jenkins-jfrog-plugin");
```
Comment (yahavi) [inline src/main/java/io/jenkins/plugins/jfrog/JfrogInstallation.java:66]: The user agent should contain the version too (add null checks):
```java
Jenkins.getInstanceOrNull().getPlugin("jfrog").getWrapper().getVersion()
```

File: src/main/java/io/jenkins/plugins/jfrog/JfrogInstallation.java [hunk 5]
```diff
@@ -49,6 +54,17 @@ public void buildEnvVars(EnvVars env) {
...
+        }
```
Comment (yahavi) [inline src/main/java/io/jenkins/plugins/jfrog/JfrogInstallation.java:67]:
```suggestion
        env.putIfAbsent(JFROG_CLI_USER_AGENT, "jenkins-jfrog-plugin");
```

**Review by yahavi — CHANGES_REQUESTED** [review]

**Pull request merged**